minard/views.py

```diff
@@ -216,8 +216,33 @@ def detector_state_check(run=None):
 @app.route('/channel-database')
 def channel_database():
     limit = request.args.get("limit", 100, type=int)
+    pmt_type = request.args.get("pmt_type", -1, type=int)
     results = get_channels(request.args, limit)
-    return render_template('channel_database.html', results=results, limit=limit)
+
+    active_type = pmt_type | 0x1
+    pmt_info = get_pmt_types()
+
+    # Allow sorting by PMT Type and apply the limit
+    row_count = 0
+    new_results = []
+    for row in results:
+        crate = row['crate']
+        slot = row['slot']
+        channel = row['channel']
+        if pmt_type == -1:
+            pass
+        elif pmt_info[crate][slot][channel] != pmt_type and \
+             pmt_info[crate][slot][channel] != active_type:
+            continue
+        row_count+=1
+        status = filter_channel_status(row)
+        row['status'] = status
+        row['pmt_type'] = pmt_type_description(pmt_info[crate][slot][channel])
+        new_results.append(row)
+        if row_count >= limit:
+            break
```

Comment:
This works, but seems a little hacky. I wonder if it would be better to add all this to the get_channels function by doing a join in the query, i.e.

```sql
SELECT * FROM current_channel_status, pmt_info WHERE current_channel_status.crate = pmt_info.crate AND current_channel_status.slot = pmt_info.slot AND current_channel_status.channel = pmt_info.channel
```

That way you should just be able to pass the type parameter to the get_channels function and it will all just work.

Reply:
Thanks, that was quite helpful, the new implementation should be much less hacky.